ballerina/http_response.bal

```diff
@@ -539,6 +539,30 @@ public class Response {
         }
         return cookiesInResponse;
     }
+
+    isolated function performDataBinding(typedesc<anydata> targetType, boolean requireValidation) returns anydata|ClientError {
+        anydata payload = check performDataBinding(self, targetType);
+        if requireValidation {
+            return performDataValidation(payload, targetType);
+        }
+        return payload;
+    }
+
+    isolated function getApplicationResponseError() returns ClientError {
+        string reasonPhrase = self.reasonPhrase;
+        map<string[]> headers = getHeaders(self);
+        anydata|error payload = getPayload(self);
+        int statusCode = self.statusCode;
+        if payload is error {
+            if payload is NoContentError {
+                return createResponseError(statusCode, reasonPhrase, headers);
+            }
+            return error PayloadBindingClientError("http:ApplicationResponseError creation failed: " + statusCode.toString() +
```

Comment:
Shall we use string template to construct the error message ?

Reply:
Fixed via [bce164f](https://github.com/ballerina-platform/module-ballerina-http/pull/1915/commits/bce164f18e1e33b548e2db2aa20f294e437a2202)